### PR TITLE
feat: expose native SDK initialize methods as new `initialize` API

### PR DESCRIPTION
**Pull request by dylancom**
### Description

It has been adviced to initialize the sdk before loading ads. (Especially when using mediation).
Therefore these methods should be exposed to RN.

Solves: #4

### CODE REVIEW

**Comment by mikehardy** [comment]
haha now the objective-c there is no escaping the lint checks. But you can run them locally for fast cycles...

File: lib/index.d.ts [hunk 1]
```diff
@@ -725,6 +725,17 @@ export namespace GoogleAdsTypes {
...
+  /**
+   * Initialization status of each ad network available to the Google Mobile Ads SDK, keyed by its
+   * GADMAdapter's class name. The list of available ad networks may be incomplete during early
+   * phases of SDK initialization.
+   */
+  export type InitializationStatus = {
+    name: string;
+    description: string;
+    status: 0 | 1;
+  }[];
```
Comment (mikehardy) [inline lib/index.d.ts:737]: An interesting choice to make the type itself an array. Typically the type is not the array, and the return is an array of the type ? Seems more ergonomic to me, as then you can address each array entry as the type, as opposed to some part of the type ?

Comment (mikehardy) [reply lib/index.d.ts:737]: Then the name is maybe not excellent, perhaps GADMAdapterInitializationStatus ? A bit unwieldy, but that captures what it *is* I think?

Comment (dylancom) [reply lib/index.d.ts:737]: Agree on the ergonomic choice. Will call it "AdapterStatus", that will align with how the types are called in iOS / Android.

**Review by mikehardy — APPROVED** [review]
Nice! I like it a lot! Mike+Elliot @ Invertase loved the idea of having you get a real role (literally I quote "the more the merrier" :-) ), so for maintainer purposes my style on these now is to label them "pending merge", which means as soon as CI goes green *anyone* with the capability can merge so long as they pay careful attention to the PR title so it matches conventional commits (and thus drives the next release correctly). I'll add a conventional commits PR title checker action next to make that easier, but that's basically it for merges. Then go to Actions and choose Publish, do a manual run off 'main' and it ...should all work :-)

**Comment by dylancom** [comment]
@mikehardy awesome 🎉! Had no time anymore yesterday evening to try it out, but thanks for the explanation :)

**Pull request merged**